internal/provider/oauth.go

```diff
@@ -91,6 +91,8 @@ func requestNewSTSOAuthToken(ctx context.Context, subjectToken, identityPoolId,
 		req = req.ExpiresIn(int32(expiredInSecondsInt))
 	}
 
+	tflog.Debug(ctx, "requesting new STS OAuth token")
```

Comment:
nit: Do we want to include other debug information to make these logs more actionable? For example, should we add the reason for the refresh, etc.?